tests/test_non_linearity_correction.py

```diff
@@ -0,0 +1,154 @@
+#A file to test the non-linearity correction, including a comparison with the II&T pipeline
+import os
+import glob
+import numpy as np
+import corgidrp.mocks as mocks
+import corgidrp.data as data
+import corgidrp.l1_to_l2a as l1_to_l2a
+from scipy import interpolate
+import pytest
+
+class NonlinException(Exception):
+    """Exception class for nonlin module."""
+
+def _parse_file(nonlin_path):
+    """
+    Get data from nonlin file.
+    
+    
+    Args:
+        nonlin_path (str): (Optional) Full path to the non-linearity calibrationf ile.
+
+    Returns:
+        tuple:
+            gain_ax (numpy.array): The gain axis
+            count_ax (numpy.array): The count axis
+            relgains (numpy.array): The relative gains
+    """
+    # Read nonlin csv
+    nonlin_raw = np.genfromtxt(nonlin_path, delimiter=',')
+
+    # File format checks
+    if nonlin_raw.ndim < 2 or nonlin_raw.shape[0] < 2 or \
+       nonlin_raw.shape[1] < 2:
+        raise NonlinException('Nonlin array must be at least 2x2 (room for x '
+                              'and y axes and one data point)')
+    if not np.isnan(nonlin_raw[0, 0]):
+        raise NonlinException('First value of csv (upper left) must be set to '
+                              '"nan"')
+
+    # Column headers are gains, row headers are dn counts
+    gain_ax = nonlin_raw[0, 1:]
+    count_ax = nonlin_raw[1:, 0]
+    # Array is relative gain values at a given dn count and gain
+    relgains = nonlin_raw[1:, 1:]
+
+    # Check for increasing axes
+    if np.any(np.diff(gain_ax) <= 0):
+        raise NonlinException('Gain axis (column headers) must be increasing')
+    if np.any(np.diff(count_ax) <= 0):
+        raise NonlinException('Counts axis (row headers) must be increasing')
+    # Check that curves (data in columns) contain or straddle 1.0
+    if (np.min(relgains, axis=0) > 1).any() or \
+       (np.max(relgains, axis=0) < 1).any():
+        raise NonlinException('Gain curves (array columns) must contain or '
+                              'straddle a relative gain of 1.0')
+
+    return gain_ax, count_ax, relgains
+
+
+def get_relgains(frame, em_gain, nonlin_path):
```

Comment:
It would be useful to include in the docstring that this is an exact copy of the II&T pipeline code.

Reply:
Done